FIAT/reference_element.py

```diff
@@ -483,6 +485,16 @@ def __init__(self):
         super(Point, self).__init__(POINT, verts, topology)
 
 
+class UFCPoint(UFCSimplex):
+    """This is the reference point with the properties of a
+    `UFCSimplex`."""
+
+    def __init__(self):
+        verts = ((),)
+        topology = {0: {0: (0,)}}
+        super(UFCPoint, self).__init__(POINT, verts, topology)
```

Comment:
This `UFCPoint` isn't any difference than `Point` above, so what's the point?

Reply:
It needs to inherit the methods of `UFCSimplex` which plain `Simplex`es don't define (this is what I refer to when I write `with the properties of a `UFCSimplex`.` in the docstring) . See the snippet below:

```python
class UFCSimplex(Simplex):

    def get_facet_element(self):
        dimension = self.get_spatial_dimension()
        return self.construct_subelement(dimension - 1)

    def construct_subelement(self, dimension):
        """Constructs the reference element of a cell subentity
        specified by subelement dimension.

        :arg dimension: subentity dimension (integer)
        """
        return ufc_simplex(dimension)

    def contains_point(self, point, epsilon=0):
        """Checks if reference cell contains given point
        (with numerical tolerance)."""
        result = (sum(point) - epsilon <= 1)
        for c in point:
            result &= (c + epsilon >= 0)
        return result
```

It just so happens that the so-called `UFCSimplex` for a point is defined identically to the plain-old `Simplex` case. I don't really know how to get around the fact the `Simplex` and `UFCSimplex` don't have identical method definitions without moving the methods out of `UFCSimplex` and into its child classes. I also don't see how to avoid repeating code for the `UFCSimplex` case given the different class hierarchies.

Reply:
None of the additional methods of `UFCSimplex` seem very relevant for points: points do not have facets, they do not have proper subcells (its only subcell is the cell itself), and `contains_point` is also quite irrelevant given that points do not really have coordinates.

Reply:
I do, however, need `construct_subelement` which tsfc uses.

Reply:
In that case, just augment the existing `Point` class with:
```python
def construct_subelement(self, dimension):
    assert dimension == 0
    return self
```

Reply:
So then `Point` becomes the only simplex that defines that method, which is repeated code from `UFCSimplex`? How is that any better than the current solution which has the advantage of maintaining consistency with the existing code?

Reply:
I decided I agreed with you and made the necessary changes in bfb17119e593c9c6d303ccbb55cb6d5d12e28725 and 04d6f17bfa8f91c00c942ca0c06d6c6691cdc1fd